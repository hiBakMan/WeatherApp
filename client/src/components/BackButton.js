const BackButton = ({setCurrentWeather}) => {
    return (
       <button data-testid="back-btn" className="back-btn" onClick={() => setCurrentWeather('')}>
       Back to List of Locations    
       </button>
    )   
};

export default BackButton;