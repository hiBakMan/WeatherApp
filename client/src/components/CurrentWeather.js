import { capitalize, formatDateTime, weekdayDateTime, getTime } from '../auxiliary/helperfunctions'

const CurrentWeather = ({currentWeather, location}) => {
    return (
        <div data-testid="curwea-box" className="curwea-box">
            <div className="location">{location}</div>
            <div className="date">{formatDateTime(currentWeather.time)}</div>
            <div className="weekday">{weekdayDateTime(currentWeather.time)}</div>
            <div className="time">{getTime(currentWeather.time)}</div>
            <div className="description">{capitalize(currentWeather.symbolPhrase)}</div>
            <div className="temperature">Temperature: {currentWeather.temperature} °C</div>
            <div className="feels">Feels Like: {currentWeather.feelsLikeTemp} °C</div>
            <div className="winddir">Wind Direction: {currentWeather.windDirString}</div>
            <div className="windspeed">Wind Speed: {currentWeather.windSpeed} m/s</div>
            <img src={process.env.PUBLIC_URL + `/assets/${currentWeather.symbol}.png`} alt="weather symbol"></img>
        </div>
    )   
};


export default CurrentWeather;

