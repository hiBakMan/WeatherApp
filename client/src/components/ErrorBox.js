const ErrorBox = ({error}) => {
    return (
    <div data-testid="error-box" className="error-box">
      {(error === "content") ? 
        (<div>
            Allowed queries: characters and spaces up to 30 symbols in total
        </div>) :
        (<div>
            No results found for '{error}'
        </div>)
        }
    </div>
    )   
};

export default ErrorBox;