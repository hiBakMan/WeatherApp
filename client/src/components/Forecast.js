import { capitalize, formatDateTime, weekdayDateTime } from '../auxiliary/helperfunctions'

const Forecast = ({forecast}) => {

    return (
        <div key={forecast.date} data-testid="forecast-box" className="forecast-box">
            <div className="date">{formatDateTime(forecast.date)}</div>
            <div className="weekday">{weekdayDateTime(forecast.date)}</div>
            <div className="description">{capitalize(forecast.symbolPhrase)}</div>
            <div className="temperature">Temperature: {forecast.minTemp}...{forecast.maxTemp} °C</div>
            <div className="feels">Feels Like: {forecast.minFeelsLikeTemp}...{forecast.maxFeelsLikeTemp} °C</div>
            <div className="windspeed">Max Wind Speed: {forecast.maxWindSpeed} m/s</div>
            <img src={process.env.PUBLIC_URL + `/assets/${forecast.symbol}.png`} alt="weather symbol"/>
        </div>
    )   
};

export default Forecast;