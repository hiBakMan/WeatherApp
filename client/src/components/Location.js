const Location = ({location, getWeather, setCurrentWeather, setForecasts, getForecasts, postWeather, setLocation}) => {
    return (
        <button key={location.id} data-testid="location-btn" className="location-btn" onClick={() => {getWeather(location, setCurrentWeather, postWeather, setLocation); getForecasts(location, setForecasts)}}> 
        {location.name}, {location.country}
        </button>
    )   
};

export default Location;