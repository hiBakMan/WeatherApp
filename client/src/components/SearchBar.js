const SearchBar = ({query, setQuery, setLocations, getLocations, setCurrentWeather, setError, postQuery}) => {
    return (
          <input 
          className="search-bar" 
          data-testid="search-bar"
          type="text" 
          placeholder="Please enter a city name..." 
          onChange={e => setQuery(e.target.value)}
          value={query}
          onKeyPress={e => {getLocations(e, query, setLocations, setQuery, setCurrentWeather, setError); postQuery(e, query)}}
          />
)
};

export default SearchBar;