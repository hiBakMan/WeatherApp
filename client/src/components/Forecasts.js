import Forecast from "./Forecast";

const Forecasts = ({forecasts}) => {
    forecasts = forecasts.slice(1, forecasts.length + 1);
    
    return (
        <div className="forecasts-box" data-testid="forecasts-box">
        {forecasts.map((forecast) => {
        return (
        <Forecast 
        key={forecast.date} 
        forecast={forecast} 
        />
        )
        })}
        </div>

    )   
};

export default Forecasts;