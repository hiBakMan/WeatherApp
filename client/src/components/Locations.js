import Location from './Location'

const Locations = ({locations, getWeather, setCurrentWeather, setForecasts, getForecasts, postWeather, setLocation}) => {
    if (locations.length > 5) {
        locations = locations.slice(0, 5)
    };
    return (
        <div className="locations-box" data-testid="locations-box">
            {locations.map((location) => {
            return (
                <Location 
                key={location.id} 
                location={location} 
                getWeather={getWeather} 
                setCurrentWeather={setCurrentWeather}
                setForecasts={setForecasts}
                getForecasts={getForecasts}
                postWeather={postWeather}
                setLocation={setLocation}
                />
            )
            })}
        </div>
    )
};

export default Locations;