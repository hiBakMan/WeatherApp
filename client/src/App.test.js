import React from 'react';
import ReactDOM from 'react-dom'
import { render, cleanup, fireEvent } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer"

import BackButton from "./components/BackButton";
import CurrentWeather from "./components/CurrentWeather";
import ErrorBox from './components/ErrorBox';
import Forecast from './components/Forecast'
import Forecasts from "./components/Forecasts";
import Location from './components/Location'
import Locations from "./components/Locations";
import SearchBar from "./components/SearchBar";
import SearchBox from "./components/SearchBox";
import SearchButton from "./components/SearchButton";



// Back Button Testing


describe('Back Button', () => {

afterEach(cleanup);

test('is rendered without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<BackButton></BackButton>, div);
})

test('is rendered correctly', () => {
    const { getByTestId } = render(<BackButton></BackButton>);
    expect(getByTestId('back-btn')).toHaveTextContent("Back to List of Locations");
})

test('matches snapshot', () => {
    const tree = renderer.create(<BackButton>Back to List of Locations</BackButton>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('has OnClick triggering a function', () => {
    const testFun = jest.fn();
    const { getByTestId } = render(<BackButton setCurrentWeather={testFun}></BackButton>);
    fireEvent.click(getByTestId('back-btn'));
    expect(testFun).toHaveBeenCalled();
})

});


// Current Weather Box testing

describe('Current Weather Box', () => {

    afterEach(cleanup);
    
    test('is rendered without crashing', () => {
        const testLoc = "Test City, Test Country"
        const testWeather = {
            time: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            temperature: "1",
            feelsLikeTemp: "-3",
            windDirString: "NW",
            windSpeed: "5",
            symbol: "d300"
        }


        const div = document.createElement("div");
        ReactDOM.render(<CurrentWeather currentWeather={testWeather} location={testLoc} ></CurrentWeather>, div);
    })
    
    test('is rendered correctly', () => {
        const testLoc = "Test City, Test Country"
        const testWeather = {
            time: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            temperature: "1",
            feelsLikeTemp: "-3",
            windDirString: "NW",
            windSpeed: "5",
            symbol: "d300"
        }
        const { getByTestId } = render(<CurrentWeather currentWeather={testWeather} location={testLoc}></CurrentWeather>);
        expect(getByTestId('curwea-box')).toHaveTextContent("Test City, Test Country8 February 2022Tuesday12:00:00 PMCLOUDYTemperature: 1 °CFeels Like: -3 °CWind Direction: NWWind Speed: 5 m/s");
    })
    
    test('matches snapshot', () => {
        const testLoc = "Test City, Test Country"
        const testWeather = {
            time: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            temperature: "1",
            feelsLikeTemp: "-3",
            windDirString: "NW",
            windSpeed: "5",
            symbol: "d300"
        }
        const tree = renderer.create(<CurrentWeather currentWeather={testWeather} location={testLoc}></CurrentWeather>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    });

// Error Box Testing


describe('Error Box', () => {

    afterEach(cleanup);
    
    test('is rendered without crashing', () => {

        const error = ""

        const div = document.createElement("div");
        ReactDOM.render(<ErrorBox error={error}></ErrorBox>, div);
    })
    
    test('is rendered correctly when content error exists', () => {
        const error = "content"

        const { getByTestId } = render(<ErrorBox error={error}></ErrorBox>);
        expect(getByTestId('error-box')).toHaveTextContent("Allowed queries: characters and spaces up to 30 symbols in total");
    })
    
    test('is rendered correctly when other error exists', () => {
        
        const error = "test"

        const { getByTestId } = render(<ErrorBox error={error}></ErrorBox>);
        expect(getByTestId('error-box')).toHaveTextContent("No results found for 'test'");
    })

    test('matches snapshot', () => {

        const error = "";

        const tree = renderer.create(<ErrorBox error={error}></ErrorBox>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    });
  
// Forecast Box testing

describe('Forecast Box', () => {

    afterEach(cleanup);
    
    test('is rendered without crashing', () => {

        const testFor = {
            date: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            minTemp: "-1",
            maxTemp: "1",
            minFeelsLikeTemp: "-3",
            minFeelsLikeTemp: "3",
            maxWindSpeed: "5",
            symbol: "d300"
        }


        const div = document.createElement("div");
        ReactDOM.render(<Forecast forecast={testFor} ></Forecast>, div);
    })
    
    test('is rendered correctly', () => {
       
        const testFor = {
            date: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            minTemp: "-1",
            maxTemp: "1",
            minFeelsLikeTemp: "-3",
            minFeelsLikeTemp: "3",
            maxWindSpeed: "5",
            symbol: "d300"
        }
        const { getByTestId } = render(<Forecast forecast={testFor}></Forecast>);
        expect(getByTestId('forecast-box')).toHaveTextContent("8 February 2022TuesdayCLOUDYTemperature: -1...1 °CFeels Like: 3... °CMax Wind Speed: 5 m/s");
    })
    
    test('matches snapshot', () => {

        const testFor = {
            date: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            minTemp: "-1",
            maxTemp: "1",
            minFeelsLikeTemp: "-3",
            minFeelsLikeTemp: "3",
            maxWindSpeed: "5",
            symbol: "d300"
        }
        const tree = renderer.create(<Forecast forecast={testFor}></Forecast>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    });

// Forecasts Box testing

describe('Forecasts Box', () => {

    afterEach(cleanup);
    
    test('is rendered without crashing', () => {

        const testFor = [{
            date: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            minTemp: "-1",
            maxTemp: "1",
            minFeelsLikeTemp: "-3",
            minFeelsLikeTemp: "3",
            maxWindSpeed: "5",
            symbol: "d300"
        }];


        const div = document.createElement("div");
        ReactDOM.render(<Forecasts forecasts={testFor} ></Forecasts>, div);
    })
    
    test('is rendered correctly with 3 forecasts', () => {
       
        const testFor = [{
            date: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            minTemp: "-1",
            maxTemp: "1",
            minFeelsLikeTemp: "-3",
            minFeelsLikeTemp: "3",
            maxWindSpeed: "5",
            symbol: "d300"
        },
        {
            date: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            minTemp: "-1",
            maxTemp: "1",
            minFeelsLikeTemp: "-3",
            minFeelsLikeTemp: "3",
            maxWindSpeed: "5",
            symbol: "d300"
        },
        {
            date: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            minTemp: "-1",
            maxTemp: "1",
            minFeelsLikeTemp: "-3",
            minFeelsLikeTemp: "3",
            maxWindSpeed: "5",
            symbol: "d300"
        }];

        const { getByTestId } = render(<Forecasts forecasts={testFor}></Forecasts>);
        expect(getByTestId('forecasts-box')).toHaveTextContent("8 February 2022TuesdayCLOUDYTemperature: -1...1 °CFeels Like: 3... °CMax Wind Speed: 5 m/s8 February 2022TuesdayCLOUDYTemperature: -1...1 °CFeels Like: 3... °CMax Wind Speed: 5 m/s");
    })
    
    test('matches snapshot', () => {

        const testFor = [{
            date: "Tue Feb 08 2022 12:00:00 GMT+0200 (Eastern European Standard Time)",
            symbolPhrase: "Cloudy",
            minTemp: "-1",
            maxTemp: "1",
            minFeelsLikeTemp: "-3",
            minFeelsLikeTemp: "3",
            maxWindSpeed: "5",
            symbol: "d300"
        }]
        const tree = renderer.create(<Forecasts forecasts={testFor}></Forecasts>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    });


// Location Button Testing


describe('Location Button', () => {

    afterEach(cleanup);
    
    test('is rendered without crashing', () => {
        const testLoc = {
            name: "Test City",
            country: "Test Country"
        };


        const div = document.createElement("div");
        ReactDOM.render(<Location location={testLoc}></Location>, div);
    })
    
    test('is rendered correctly', () => {
        const testLoc = {
            name: "Test City",
            country: "Test Country"
        };

        const { getByTestId } = render(<Location location={testLoc}></Location>);
        expect(getByTestId('location-btn')).toHaveTextContent("Test City, Test Country");
    })
    
    test('matches snapshot', () => {
        const testLoc = {
            name: "Test City",
            country: "Test Country"
        };

        const tree = renderer.create(<Location location={testLoc}>Test City, Test Country</Location>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    test('has OnClick triggering the first function', () => {
        const testLoc = {
            name: "Test City",
            country: "Test Country"
        };

        const testFunFirst = jest.fn();
        const testFunSecond = jest.fn();

        const { getByTestId } = render(<Location location={testLoc} getWeather={testFunFirst} getForecasts={testFunSecond}></Location>);
        fireEvent.click(getByTestId('location-btn'));
        expect(testFunFirst).toHaveBeenCalled();
    })

    test('has OnClick triggering the second function', () => {
        const testLoc = {
            name: "Test City",
            country: "Test Country"
        };

        const testFunFirst = jest.fn();
        const testFunSecond = jest.fn();

        const { getByTestId } = render(<Location location={testLoc} getWeather={testFunFirst} getForecasts={testFunSecond}></Location>);
        fireEvent.click(getByTestId('location-btn'));
        expect(testFunSecond).toHaveBeenCalled();
    })
    
    });

// Locations Box testing

    describe('Locations Box', () => {

        afterEach(cleanup);
        
        test('is rendered without crashing', () => {
    
            const testLocs = [{
                id: "1",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "2",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "3",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "4",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "5",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "6",
                name: "Test City",
                country: "Test Country"
            }];
    
    
            const div = document.createElement("div");
            ReactDOM.render(<Locations locations={testLocs} ></Locations>, div);
        })
        
        test('is rendered correctly with 6 locations (should render a maximum of 5)', () => {
           
            const testLocs = [{
                id: "1",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "2",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "3",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "4",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "5",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "6",
                name: "Test City",
                country: "Test Country"
            }];
    
            const { getByTestId } = render(<Locations locations={testLocs}></Locations>);
            expect(getByTestId('locations-box')).toHaveTextContent("Test City, Test CountryTest City, Test CountryTest City, Test CountryTest City, Test CountryTest City, Test Country");
        })
        
        test('matches snapshot', () => {
    
            const testLocs = [{
                id: "1",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "2",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "3",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "4",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "5",
                name: "Test City",
                country: "Test Country"
            },
            {
                id: "6",
                name: "Test City",
                country: "Test Country"
            }];

            const tree = renderer.create(<Locations locations={testLocs}></Locations>).toJSON();
            expect(tree).toMatchSnapshot();
        })
        
        });

// Search Bar testing

describe('Search Bar', () => {

    afterEach(cleanup);
    
    test('is rendered without crashing', () => {
    
        const div = document.createElement("div");
        ReactDOM.render(<SearchBar></SearchBar>, div);
    })
    
    test('is rendered correctly', () => {

        const value = "test";
       
    
        const { getByTestId } = render(<SearchBar value={value}></SearchBar>);
        expect(getByTestId('search-bar')).toHaveTextContent("");
    })
    
    test('matches snapshot', () => {

        const tree = renderer.create(<SearchBar></SearchBar>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('has OnKeyPress triggering the first function', () => {

        const value = "test";
    
        const testFunFirst = jest.fn();
        const testFunSecond = jest.fn();

        const { getByTestId } = render(<SearchBar value={value} getLocations={testFunFirst} postQuery={testFunSecond}></SearchBar>);
        fireEvent.keyPress(getByTestId('search-bar'), {key: "Enter", code: "Enter", charCode: 13});
        expect(testFunFirst).toHaveBeenCalled();
    })

    test('has OnKeyPress triggering the second function', () => {

        const value = "test";
    
        const testFunFirst = jest.fn();
        const testFunSecond = jest.fn();

        const { getByTestId } = render(<SearchBar value={value} getLocations={testFunFirst} postQuery={testFunSecond}></SearchBar>);
        fireEvent.keyPress(getByTestId('search-bar'), {key: "Enter", code: "Enter", charCode: 13});
        expect(testFunSecond).toHaveBeenCalled();
    })
    
    test('has OnChange triggering the function', () => {

        const value = "test";
    
        const testFunFirst = jest.fn();

        const { getByTestId } = render(<SearchBar value={value} setQuery={testFunFirst}></SearchBar>);
        fireEvent.change(getByTestId('search-bar'), {target: {value: "a"}});
        expect(testFunFirst).toHaveBeenCalled();
    })

    });

// Search Box testing

describe('Search Box', () => {

    afterEach(cleanup);
    
    test('is rendered without crashing', () => {
    
        const div = document.createElement("div");
        ReactDOM.render(<SearchBox></SearchBox>, div);
    })
    
    test('is rendered correctly', () => {

        const value = "test";

       
        const { getByTestId } = render(<SearchBox query={value}></SearchBox>);
        expect(getByTestId('search-box')).toHaveTextContent("Search");
    })
    
    test('matches snapshot', () => {
        
        const testQuer = "test"

        const tree = renderer.create(<SearchBox query={testQuer}></SearchBox>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    });

// Search Button testing

describe('Search Button', () => {

    afterEach(cleanup);
    
    test('is rendered without crashing', () => {
    
        const div = document.createElement("div");
        ReactDOM.render(<SearchButton></SearchButton>, div);
    })
    
    test('is rendered correctly', () => {
       
        const { getByTestId } = render(<SearchButton></SearchButton>);
        expect(getByTestId('search-btn')).toHaveTextContent("Search");
    })
    
    test('matches snapshot', () => {

        const tree = renderer.create(<SearchButton></SearchButton>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('has OnClick triggering the first function', () => {
    
        const testFunFirst = jest.fn();
        const testFunSecond = jest.fn();

        const { getByTestId } = render(<SearchButton getLocations={testFunFirst} postQuery={testFunSecond}></SearchButton>);
        fireEvent.click(getByTestId('search-btn'));
        expect(testFunFirst).toHaveBeenCalled();
    })

    test('has OnClick triggering the second function', () => {
    
        const testFunFirst = jest.fn();
        const testFunSecond = jest.fn();

        const { getByTestId } = render(<SearchButton getLocations={testFunFirst} postQuery={testFunSecond}></SearchButton>);
        fireEvent.click(getByTestId('search-btn'));
        expect(testFunSecond).toHaveBeenCalled();
    })
    
    });
