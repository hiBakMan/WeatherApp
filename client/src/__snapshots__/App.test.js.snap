// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Button matches snapshot 1`] = `
<button
  className="back-btn"
  data-testid="back-btn"
  onClick={[Function]}
>
  Back to List of Locations
</button>
`;

exports[`Current Weather Box matches snapshot 1`] = `
<div
  className="curwea-box"
  data-testid="curwea-box"
>
  <div
    className="location"
  >
    Test City, Test Country
  </div>
  <div
    className="date"
  >
    8 February 2022
  </div>
  <div
    className="weekday"
  >
    Tuesday
  </div>
  <div
    className="time"
  >
    12:00:00 PM
  </div>
  <div
    className="description"
  >
    CLOUDY
  </div>
  <div
    className="temperature"
  >
    Temperature: 
    1
     째C
  </div>
  <div
    className="feels"
  >
    Feels Like: 
    -3
     째C
  </div>
  <div
    className="winddir"
  >
    Wind Direction: 
    NW
  </div>
  <div
    className="windspeed"
  >
    Wind Speed: 
    5
     m/s
  </div>
  <img
    alt="weather symbol"
    src="/assets/d300.png"
  />
</div>
`;

exports[`Error Box matches snapshot 1`] = `
<div
  className="error-box"
  data-testid="error-box"
>
  <div>
    No results found for '
    
    '
  </div>
</div>
`;

exports[`Forecast Box matches snapshot 1`] = `
<div
  className="forecast-box"
  data-testid="forecast-box"
>
  <div
    className="date"
  >
    8 February 2022
  </div>
  <div
    className="weekday"
  >
    Tuesday
  </div>
  <div
    className="description"
  >
    CLOUDY
  </div>
  <div
    className="temperature"
  >
    Temperature: 
    -1
    ...
    1
     째C
  </div>
  <div
    className="feels"
  >
    Feels Like: 
    3
    ...
     째C
  </div>
  <div
    className="windspeed"
  >
    Max Wind Speed: 
    5
     m/s
  </div>
  <img
    alt="weather symbol"
    src="/assets/d300.png"
  />
</div>
`;

exports[`Forecasts Box matches snapshot 1`] = `
<div
  className="forecasts-box"
  data-testid="forecasts-box"
/>
`;

exports[`Location Button matches snapshot 1`] = `
<button
  className="location-btn"
  data-testid="location-btn"
  onClick={[Function]}
>
  Test City
  , 
  Test Country
</button>
`;

exports[`Locations Box matches snapshot 1`] = `
<div
  className="locations-box"
  data-testid="locations-box"
>
  <button
    className="location-btn"
    data-testid="location-btn"
    onClick={[Function]}
  >
    Test City
    , 
    Test Country
  </button>
  <button
    className="location-btn"
    data-testid="location-btn"
    onClick={[Function]}
  >
    Test City
    , 
    Test Country
  </button>
  <button
    className="location-btn"
    data-testid="location-btn"
    onClick={[Function]}
  >
    Test City
    , 
    Test Country
  </button>
  <button
    className="location-btn"
    data-testid="location-btn"
    onClick={[Function]}
  >
    Test City
    , 
    Test Country
  </button>
  <button
    className="location-btn"
    data-testid="location-btn"
    onClick={[Function]}
  >
    Test City
    , 
    Test Country
  </button>
</div>
`;

exports[`Search Bar matches snapshot 1`] = `
<input
  className="search-bar"
  data-testid="search-bar"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Please enter a city name..."
  type="text"
/>
`;

exports[`Search Box matches snapshot 1`] = `
<div
  className="search-box"
  data-testid="search-box"
>
  <input
    className="search-bar"
    data-testid="search-bar"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Please enter a city name..."
    type="text"
    value="test"
  />
  <button
    className="search-btn"
    data-testid="search-btn"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;

exports[`Search Button matches snapshot 1`] = `
<button
  className="search-btn"
  data-testid="search-btn"
  onClick={[Function]}
>
  Search
</button>
`;
